test/distributed/org/apache/cassandra/distributed/test/cql3/StatefulASTBase.java

```diff
@@ -485,6 +505,21 @@ protected boolean allowUsingTimestamp()
             return true;
         }
 
+        protected final boolean allowTxn()
+        {
+            Invariants.nonNull(metadata, "During object setup please use allowTxn(TableMetadata metadata)");
+            return allowTxn(metadata);
+        }
+
+        /**
+         * Should BEGIN TRANSACTION queries be generated. As of this moment BEGIN TRANSACTION does not support USING TIMESTAMP so it's unsafe for
+         * {@link #allowTxn(TableMetadata)} and {@link #allowUsingTimestamp()} to both be true.
+         */
+        protected boolean allowTxn(TableMetadata metadata)
+        {
+            return metadata.params.transactionalMode.accordIsEnabled && !allowUsingTimestamp();
+        }
+
```

Comment:
this is actually from [CASSANDRA-20828](https://issues.apache.org/jira/browse/CASSANDRA-20828) but was easier for me to keep than remove.